pkg/proc/gdbserial/gdbserver.go

```diff
@@ -1926,6 +2179,10 @@ func (t *gdbThread) SetSP(sp uint64) error {
 
 // SetDX will set the value of the DX register to the given value.
 func (t *gdbThread) SetDX(dx uint64) error {
+	if runtime.GOARCH == "arm64" {
```

Comment:
Wherever you use runtime.GOARCH can you use `t.p.bi.Arch.Name`/`p.bi.Arch.Name` instead?

Reply:
Yeah sure, makes sense to use the target architecture instead of the runtime.

I am wondering whether we should consider some cornercases where one might need to debug a x86 application on an arm64 System - which theoretically should be possible, since lldb also manages to do that. I can keep investigating this once the initial PR is done

Reply:
Ideally those regnumPC constants would be member fields of gdbProcess (or gdbConn) and setup correctly on initialization, so it wouldn't need conditional compilation and (should) work across architectures. But we don't need to do this here.

Reply:
> Wherever you use runtime.GOARCH can you use `t.p.bi.Arch.Name`/`p.bi.Arch.Name` instead?

I cannot access the process in the `gdbRegisters` functions :( Such as `func (regs *gdbRegisters) TLS() uint64` any ideas?

Reply:
`gdbRegistes.setDX` is only called by `gdbThread.SetDX` so it can be done there. For TLS and GAddr instead of changing those functions you can change the final part of reloadRegisters to set `t.regs.tls`, `t.regs.hasgaddr` and `t.regs.gaddr` appropriately. That leaves `Get` for which there is no easy solution, you can change gdbRegisters.init to take a `goarch string` parameter and save it as a field of gdbRegisters and use that in Get. BTW do you mind also changing the constants in Get to this:

```
const (
    mask8 = 0xff
    mask16 = 0xffff
    mask32 = 0xffffffff
)
```

Thanks.

Reply:
I have added a *proc.Arch field to the gdbRegisters and the init function - or would you prefer a string field? TLS and GAddr are set in reloadRegisters now as you suggested

Reply:
> I have added a *proc.Arch field to the gdbRegisters and the init function - or would you prefer a string field?

either way is fine.

Reply:
> Ideally those regnumPC constants would be member fields of gdbProcess (or gdbConn) and setup correctly on initialization, so it wouldn't need conditional compilation and (should) work across architectures. But we don't need to do this here.

I tried that quickly, but then I am running into the same issue as above, with the `gdbRegisters` not able to access them if they are either in the process or in the connection. I think the best way to handle this would actually be in in `proc.Arch` since this seems the obvious choice to know what the "special" registers are on each platform. Maybe this even something that DWARF can solve? I am not an expert on this. But this probably then goes beyond this PR - it is however needed if we can to be able to debug translated rosetta2 binaries.

Reply:
I'm going to make a few changes to how we manipulate registers to support writing to variables stored in registers (#2271), it's possible that it will help, however some things here are a bit specific to gdb. I agree that it should be postponed.